### PR TITLE
source: check whether GTID is `ON` in the upstream ealier

### LINKED ISSUE #941: check whether GTID is `ON` in the upstream ealier
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

2. What did you expect to see?

3. What did you see instead?

    ```
    [2020/08/27 16:07:03.684 +08:00] [INFO] [checkpoint.go:798] ["parsing metadata from file"] [task=nmm3] [unit="binlog replication"] [component="remote checkpoint"] [file=dumped_data.nmm3/metadata]
    [2020/08/27 16:07:03.684 +08:00] [INFO] [checkpoint.go:761] ["loaded checkpoints from meta"] [task=nmm3] [unit="binlog replication"] [component="remote checkpoint"] ["global checkpoint"="position: (mysql-bin.001974, 443173257), gtid-set: (flushed position: (mysql-bin.001974, 443173257), gtid-set: )"]
    [2020/08/27 16:07:03.880 +08:00] [INFO] [syncer.go:885] ["flushed checkpoint"] [task=nmm3] [unit="binlog replication"] [checkpoint="position: (mysql-bin.001974, 443173257), gtid-set: (flushed position: (mysql-bin.001974, 443173257), gtid-set: )"]
    [2020/08/27 16:07:03.880 +08:00] [INFO] [syncer.go:1098] ["try to remove loaded files"] [task=nmm3] [unit="binlog replication"]

    [2020/08/27 16:07:03.887 +08:00] [ERROR] [syncer.go:1290] ["fail to fetch binlog"] [task=nmm3] [unit="binlog replication"] [error="ERROR 1236 (HY000): The replication sender thread cannot start in AUTO_POSITION mode: this server has GTID_MODE = OFF instead of ON."]
   ...
    [2020/08/27 16:07:03.889 +08:00] [ERROR] [subtask.go:295] ["unit process error"] [subtask=nmm3] [unit=Sync] ["error information"="{\"ErrCode\":50000,\"ErrClass\":\"not-set\",\"ErrScope\":\"not-set\",\"ErrLevel\":\"high\",\"Message\":\"ERROR 1236 (HY000): The replication sender thread cannot start in AUTO_POSITION mode: this server has GTID_MODE = OFF instead of ON.\"}"]
    [2020/08/27 16:07:06.586 +08:00] [INFO] [worker.go:288] ["auto_resume sub task"] [component="worker controller"] [task=nmm3]
    ...
    [2020/08/27 16:07:06.634 +08:00] [INFO] [checkpoint.go:798] ["parsing metadata from file"] [task=nmm3] [unit="binlog replication"] [component="remote checkpoint"] [file=dumped_data.nmm3/metadata]
    [2020/08/27 16:07:06.635 +08:00] [ERROR] [subtask.go:295] ["unit process error"] [subtask=nmm3] [unit=Sync] ["error information"="{\"ErrCode\":11001,\"ErrClass\":\"functional\",\"ErrScope\":\"internal\",\"ErrLevel\":\"high\",\"Message\":\"parse mydumper metadata error: open dumped_data.nmm3/metadata: no such file or directory\"}"]
    ```

    `query-status` report `parse mydumper metadata error: open dumped_data.nmm3/metadata: no such file or directory`, but the real cause is `ERROR 1236 (HY000): The replication sender thread cannot start in AUTO_POSITION mode: this server has GTID_MODE = OFF instead of ON.`

    [more.log](https://github.com/pingcap/dm/files/5135281/more.log)

    Can also see it at [AskTUG](https://asktug.com/t/topic/36828).

   we can check whether `GTID_MODE=on` when `operate-source create`.

   

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v2.0.0-rc
        ```

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #941 

### What is changed and how it works?
check whether GTID is `ON` in the upstream when operate-source create

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 run mysql with GTID_MODE=OFF, and set enable-gtid: true in source config

### CODE REVIEW

File: dm/config/source_config.go [hunk 1]
```diff
@@ -318,3 +318,18 @@ func (c *SourceConfig) check(metaData *toml.MetaData, err error) error {
 	c.adjust()
 	return nil
 }
+
+// PreCheck check valify source config
+func (c *SourceConfig) PreCheck(db *sql.DB) error {
```
Comment (lance6716) [inline dm/config/source_config.go:323]: `Adjust` could check and return error, so maybe we could put these code into `Adjust`?

Comment (csuzhangxc) [reply dm/config/source_config.go:323]: agree. seems we have use `*sql.DB` in `Adjust` now.

**Comment by lance6716** [comment]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**